tests/convert-song.spec.ts

```diff
@@ -33,176 +33,238 @@ const FINAL_GENRE = 'Final Genre';
-  await page.getByTestId('undo-change').click();
-  await page.getByTestId('section-3').click();
-  await expect(page.locator('[data-test="change-start-beat"] input')).toHaveValue('103'); // next section
-
-  await page.getByTestId('section-9').click();
-  await page.getByTestId('delete-section').click();
-
-  await page.getByTestId('next-button').click();
-  await page.getByTestId('previous-button').click();
-
-  await expect(page.locator('[data-test="shift-video-gap"] input')).toHaveValue('10');
-  await expect(page.locator('[data-test="shift-gap"] input')).toHaveValue('1000');
-  await expect(page.locator('[data-test="change-bpm"] input')).toHaveValue(FINAL_BPM);
-
-  await page.getByTestId('next-button').click();
-
-  // Song metadata
-  await expect(page.locator('[data-test="song-artist"] input')).toHaveValue('convert');
-  await page.locator('[data-test="song-artist"] input').fill(FINAL_ARTIST);
-
-  await expect(page.locator('[data-test="song-title"] input')).toHaveValue('test');
-  await page.locator('[data-test="song-title"] input').fill(FINAL_TITLE);
-
-  await expect(page.locator('[data-test="song-genre"] input')).toHaveValue('genre');
-  await page.locator('[data-test="song-genre"] input').fill(FINAL_GENRE);
-
-  await expect(page.locator('[data-test="song-language"]')).toContainText(FINAL_LANG[0]);
-  await page.locator('[data-test="song-language"] input').fill(FINAL_LANG[1]);
-  await page.keyboard.press('Enter');
-
-  await expect(page.locator('[data-test="release-year"] input')).toHaveValue('1992');
-  await page.locator('[data-test="release-year"] input').fill(FINAL_YEAR);
-
-  await expect(page.locator('[data-test="song-bpm"] input')).toHaveValue('200');
-  await page.locator('[data-test="song-bpm"] input').fill(FINAL_SONG_BPM);
-
-  await expect(page.locator('[data-test=song-preview] input[data-index="0"]')).toHaveValue(
-    String(+FINAL_VIDEO_GAP + 60),
-  );
-  await expect(page.locator('[data-test=song-preview] input[data-index="1"]')).toHaveValue(
-    String(+FINAL_VIDEO_GAP + 60 + 30),
-  );
-  await page.locator('[data-test=song-preview] input[data-index="0"]').fill(FINAL_PREVIEW_START);
-  await page.locator('[data-test=song-preview] input[data-index="1"]').fill(FINAL_PREVIEW_END);
-
-  await expect(page.locator('[data-test="volume"] input')).toHaveValue('0.25');
-  await page.locator('[data-test="volume"] input').fill(FINAL_VOLUME);
-
-  // Download song
-  await expect(page.getByTestId('next-button')).not.toBeVisible();
-  await expect(page.getByTestId('save-button')).toBeVisible();
-  await page.getByTestId('save-button').click();
-  await page.getByTestId('share-songs-disagree').click();
-  await expect(page.getByTestId('share-songs-switch').getByRole('checkbox')).not.toBeChecked();
-
+  });
+
+  await test.step('Toggle video gap shift controls - video gap indicates lyrics gap value', async () => {
+    const entranceVideoGapShift = '0';
+    const videoGapControls = ['+1', '+5', '+10', '-1', '-5', '-10'];
+    {
```

Comment:
since the loop is now wrapped in `test.step`, there's no need for additional `{` and `}`